Automattic/newspack-plugin
# feat(ia): add settings seo

### All Submissions:


### Changes proposed in this Pull Request:

- Add SEO wizard to Settings SEO page
- Add new hook for simple text input validation
- Add robust FE form field validation

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets i.e. `git checkout origin/feat/ia-settings-seo && npm run build`.
2. Navigate to _/wp-admin/admin.php?page=newspack-settings#/seo_
3. Observe all SEO input fields are present.

![Screenshot 2024-08-23 at 09 54 40](https://github.com/user-attachments/assets/9aa93e4f-1882-4020-bbdb-966d904d54a7)

4. Test Verification codes. Ensure values are stored and appropriate error displays when below rule(s) are violated.
   - Google rules: alpha-numeric, hyphen and underscore chars
   - Bing rules: aA-fF chars, hyphen and underscore chars
5. Test Accounts. Ensure values are stored and appropriate error displays when below rule(s) are violated.
   - X rules: Alpha numeric chars and underscores. [Max length is 15](https://charactercount.tools/twitter/#:~:text=10000%20characters-,Twitter%20handle%20max%20length,-15%20characters).
   - The rest of the account fields require a valid url.

Example of errors:
![Screenshot 2024-08-23 at 10 11 28](https://github.com/user-attachments/assets/b027e39b-d5d2-4594-98a8-95b9dbb79084)

### Other information:
